addons/mail/static/src/core/common/messaging_service.js

```diff
@@ -77,9 +77,16 @@ export class Messaging {
         this.store.internalUserGroupId = data.internalUserGroupId;
         this.store.discuss.starred.counter = data.starred_counter;
         this.store.mt_comment_id = data.mt_comment_id;
-        this.store.discuss.isActive =
-            data.menu_id === this.router.current.hash?.menu_id ||
-            this.router.hash?.action === "mail.action_discuss";
+        if (!this.store.discuss.isActive) {
+            const routerhash = this.router.current.hash;
+            if (routerhash?.action === "mail.action_discuss") {
+                this.store.discuss.isActive = true;
+            } else if (data.action_discuss_id) {
+                this.store.discuss.isActive = data.action_discuss_id === routerhash?.action;
+            } else {
+                this.store.discuss.isActive = data.menu_id === routerhash?.menu_id;
```

Comment:
should be menu_id if both are undefined (it was changed in master, I'm not sure if the issue happens in 17 though)